usersession/agent/session_agent.go

```diff
@@ -79,6 +84,63 @@ func (c *Command) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	rsp.ServeHTTP(w, r)
 }
 
+type idleTracker struct {
+	mu     sync.Mutex
+	active map[net.Conn]struct{}
+	change chan struct{}
+}
+
+var sysGetsockoptUcred = sys.GetsockoptUcred
+
+func getUcred(conn net.Conn) (*sys.Ucred, error) {
+	if uconn, ok := conn.(*net.UnixConn); ok {
+		f, err := uconn.File()
+		if err != nil {
+			return nil, err
+		}
+		defer f.Close()
+		return sysGetsockoptUcred(int(f.Fd()), sys.SOL_SOCKET, sys.SO_PEERCRED)
+	}
+	return nil, fmt.Errorf("cannot get peer credentials for connection %v", conn)
+}
+
+func (it *idleTracker) trackConn(conn net.Conn, state http.ConnState) {
+	// Perform peer credentials check
+	if state == http.StateNew {
+		ucred, err := getUcred(conn)
+		uid := uint32(sys.Geteuid())
+		if err != nil || (ucred.Uid != 0 && ucred.Uid != uid) {
```

Comment:
I would feel better about this check if we also had a spread test for this which directly tested if non-root can pass this check

Reply:
I've extended the spread test to make a curl call as the test user.  I'm not sure we've got an obvious non-`root`, non-`test` user to make use of in spread tests though.